src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DateTime.cs

```diff
@@ -484,10 +484,13 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
                 // for DateTime alone is sufficient.
                 fValid &= CheckDateColumnType(type0, args[0], errors, ref nodeToCoercedTypeMap);
 
-                var resultColumnType = binding.Features.HasFlag(Features.ConsistentOneColumnTableResult)
-                    ? ColumnName_Value
-                    : type0.GetNames(DPath.Root).Single().Name;
-                returnType = DType.CreateTable(new TypedName(DType.DateTime, resultColumnType));
+                if (fValid)
+                {
+                    var resultColumnType = binding.Features.HasFlag(Features.ConsistentOneColumnTableResult)
+                        ? ColumnName_Value
+                        : type0.GetNames(DPath.Root).Single().Name;
```

Comment:
If the datecolumntype check above failed, it's possible that the table does not have a single column, so the call to `Single` would throw an exception.